packages/core/src/lavadome.mjs

```diff
@@ -29,58 +29,62 @@ navigation.addEventListener('navigate', event => {
-
-    // get/create shadow for host (empty shadow content if there's any already)
-    const shadow = getShadow(host, opts);
-    replaceChildren(shadow);
+function defineCopy(instance, input) {
+    return async function copy() {
+        const type = 'text/plain';
+        const blob = new Blob([input], {type});
+        const data = [new ClipboardItem({[type]: blob})];
+        await write(clipboard, data);
+    }
+}
 
-    // fire every time instance is reloaded and abort loading for non-top documents
-    const iframe = loadable();
-    addEventListener(iframe, 'load', () => {
-        const ownerDoc = ownerDocument(iframe);
-        if (ownerDoc !== document) {
-            replaceChildren(shadow);
-            throw new Error(`LavaDomeCore: ` +
-                `The document to which LavaDome was originally introduced ` +
-                `must be the same as the one this instance is inserted to`);
+function defineText(instance, host, opts) {
+    return function text(input) {
+        const type = typeof input;
+        if (type !== 'string') {
+            throw new Error(
+                `LavaDomeCore: first argument must be a string, instead got ${type}`);
         }
-    });
 
-    // child of the shadow, where the secret is set, must be hardened
-    const child = hardened();
-    appendChild(shadow, child);
+        // get/create shadow for host (empty shadow content if there's any already)
+        const shadow = getShadow(host, opts);
+        replaceChildren(shadow);
 
-    function text(text) {
-        if (typeof text !== 'string') {
-            throw new Error(
-                `LavaDomeCore: first argument must be a string, instead got ${stringify(text)}`);
-        }
+        // child of the shadow, where the secret is set, must be hardened
+        const child = hardened();
+        appendChild(shadow, child);
 
         // check if text is a single char and if so, either is part of a longer secret
         // which is protected by the parent LavaDome, or simply a single char provided by
         // consumer either way - not worth attempting to secure
-        if (at(from(text), 1) === undefined) {
-            return textContentSet(child, text);
+        if (at(from(input), 1) === undefined) {
+            return textContentSet(child, input);
         }
 
-        // attach loadable only once per instance to avoid excessive load firing
-        appendChild(shadow, iframe);
-
         // place each char of the secret in its own LavaDome protection instance
-        map(from(text), char => {
+        map(from(input), char => {
             const span = createElement(document, 'span');
             // mark as internal LavaDome instance
             opts[OPTIONS.isInnerInstance] = true;
             new LavaDome(span, opts).text(char);
             appendChild(child, span);
         });
 
+        // make exported API tamper-proof
+        defineProperties(instance, {
+            copy: {value: defineCopy(instance, input)},
```

Comment:
Won't this error with `Cannot redefine property: copy` if `text` is called twice on the same instance? Could we add testing coverage for what happens in that scenario?

Reply:
outdated, continue @ https://github.com/LavaMoat/LavaDome/pull/50#discussion_r1793335623